src/fairseq2/datasets/huggingface.py

```diff
@@ -18,6 +18,7 @@
     )
 except ImportError:
     has_datasets = False
+    Dataset = Any
```

Comment:
A better option would be to make the module fail fast with the explicit message for the ModuleNotFoundError (lines [81-84](https://github.com/facebookresearch/fairseq2/pull/708/files#diff-97566db29a807a8ee911a6217dfed4f0b5cc60003492927a515b874262024fb0R81-R84))

Reply:
I believed the purpose of the try-except block was to enable the CLI to perform tasks that don’t rely on the datasets library without encountering import errors. If I place the ModuleNotFoundError at the top level, it will be raised even when simply printing the help message with `fairseq2 --help`.

Reply:
We can isolate the list of tasks in fairseq2.recipes.eval.__init__._setup_eval_cli() that uses HuggingFace imports, and copy the explicitt message for the ModuleNotFoundError there.

Reply:
Something like this:

Reply:
Thank you for the guidance! I’ve implemented the suggested changes in this commit: https://github.com/facebookresearch/fairseq2/pull/708/commits/bc89baa9785f349da3bebd88b111499a910ed396